pkg/cloudinit/cloudinit.go

```diff
@@ -33,12 +35,37 @@ import (
 type YipCloudInitRunner struct {
 	exec    executor.Executor
 	fs      vfs.FS
+	v1fs    v1vfs.FS
 	console plugins.Console
 }
 
 // NewYipCloudInitRunner returns a default yip cloud init executor with the Elemental plugin set.
 // It accepts a logger which is used inside the runner.
 func NewYipCloudInitRunner(l v1.Logger, r v1.Runner, fs vfs.FS) *YipCloudInitRunner {
```

Comment:
Here is the ugly part of having to deal with different versions of the go-vfs library. Yip uses version v1, while upstream is already in v4.

Reply:
We should create a respective issue in upstream yip

Reply:
done https://github.com/mudler/yip/issues/123